### ISSUE TITLE
Reduce the size of the pangolin container?

### ISSUE THREAD

**Issue by kapsakcj** [StaPH-B/docker-builds#198]
So the Pangolin containers are up around 1GB, compressed (as listed on dockerhub).

I was wondering if there is a way to slim down the size of the container to improve download times & such.

Is it necessary to include the code from this repo? `https://github.com/cov-lineages/lineages.git`

https://github.com/StaPH-B/docker-builds/blob/2065e8a8abf4f02febfef81d32f3059b708ce8b4/pangolin/2.3.8/Dockerfile#L59-L72

Anyone have any thoughts on this? I wasn't sure if the code from this repo: `cov-lineages/lineages` was required or not.

I still want to keep the line with `conda clean -a -y` as it helps reduce the size. We could move it to a different RUN statement (but still towards the end so it's run after conda has installed things).

@andersgs @StaPH-B/docker-dev

**Comment by kapsakcj** [comment]
commenting out lines 66-72 (and moving line 72 to a previous block of commands) results in a loss of about 400MB:
```
kapsakcj/pangolin              without-lineages-repo         44ab13046545   39 seconds ago   2.22GB
staphb/pangolin                latest                        1bff26c22cb5   2 hours ago      2.61GB
```

**Comment by kapsakcj** [comment]
oh jeez....I should have just gone to the repo itself....it's archived https://github.com/cov-lineages/lineages 

🤦‍♂️ 

Sorry for pinging everyone.

**Comment by kapsakcj** [comment]
Shaved another ~10MB off by adding lines to delete pangolin tarball and pangoLEARN tarball.

**Comment by kapsakcj** [comment]
Updated image here, now at ~800MB compressed https://hub.docker.com/r/kapsakcj/pangolin/tags?page=1&ordering=last_updated

**Comment by kapsakcj** [comment]
I would be so grateful if anyone could test out the docker image I linked above.

It ran successfully on the B.1.1.7 fasta that Erin provided `tests/SARS-CoV-2/SRR13957123.consensus.fa`

**Comment by kapsakcj** [comment]
So it now occurs to me that in the pangolin-dev branch I've updated the pangolin 2.3.8 dockerfile, which has been automatically updated with new pangoLEARN version by GH actions workflow. So it's already out-of-date & not easily merged. And it may continue to be updated with new pangoLEARN versions as they are released. Not sure when pangolin itself will get a new release.

TODO:

- EITHER wait for a new version of pangolin to be released (meaning the 2.3.8 dockerfile won't be auto-updated anymore) so we could use the 2.3.8 dockerfile as the "pangolin template dockerfile" in the GH actions workflow. Then we could adjust the GH actions workflow to use this new 2.3.8 dockerfile as the template.

- OR I could instead take action now, and edit the current "pangolin template dockerfile," which is `pangolin/2.1.10/Dockerfile` with the same changes I've made to the 2.3.8 dockerfile in pangolin-dev branch. This means we wouldn't have to adjust the GH actions workflow at all.

I think I'm leaning towards the second option....plan on getting this done this week. I think it would benefit all to reduce the size of this container.
